packages/widget/src/components/address-input/address-input.ts

```diff
@@ -0,0 +1,90 @@
+import { LitElement, html } from 'lit';
+import type { HTMLTemplateResult } from 'lit';
+import { customElement, property, state } from 'lit/decorators.js';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { ethers } from 'ethers';
+import { Network } from '@buildwithsygma/sygma-sdk-core';
+import { when } from 'lit/directives/when.js';
+import { validateSubstrateAddress } from '../../utils';
+import { styles } from './styles';
+
+@customElement('sygma-address-input')
+export class AddressInput extends LitElement {
+  static styles = styles;
+  @property({
+    type: String
+  })
+  address: string = '';
+
+  @property({ attribute: false })
+  onAddressChange: (address: string) => void = () => {};
+
+  @property({
+    type: String
+  })
+  network: Network = Network.EVM;
+
+  @state()
+  errorMessage?: string;
+
+  connectedCallback(): void {
+    super.connectedCallback();
+    if (this.address) {
+      this.handleAddressChange({
+        target: { value: this.address }
+      } as unknown as Event);
```

Comment:
for this, it means we would need to do this:

```ts
 @change=${(evt: Event) =>
            this.handleAddressChange((evt.target as HTMLInputElement).value)}
```

I don't think this is hacky, is just type gymnastics sponsored by TS. Since we use this function to also validate the prefilled address upon first render of the component. Either way I think both works. What are your thoughts?

Reply:
> I don't think this is hacky, is just type gymnastics sponsored by TS.

hahahahahahaha isn't `type gimnastics` literally definition of being hacky 😆 

i'd rather unwrap event like you wrote and just take string as argument in that method, since obviously it can be called without emitting an Event

Reply:
changed